lib/pure/net.nim

```diff
@@ -678,56 +672,49 @@ when defineSsl:
     ## Sets the identity hint passed to server.
     ##
     ## Only used in PSK ciphersuites.
-    if ctx.context.SSL_CTX_use_psk_identity_hint(hint) <= 0:
+    if ctx.context.SSL_CTX_use_psk_identity_hint(hint.cstring) <= 0:
       raiseSSLError()
 
-  proc clientGetPskFunc*(ctx: SslContext): SslClientGetPskFunc =
-    return ctx.getExtraInternal().clientGetPskFunc
```

Comment:
Can we keep this? Removing it will be a breaking change plus being able to retrieve it seems useful.

Reply:
yes, that's possible.  However this was a chance to make a clean sweep and I felt freer to make 'breaking' changes as none of the psk functionality has worked since before 1.0 so i know i'm not breaking anyone's code.

Reply:
Bear in mind the pskfunc is probably declared as a proc and can be retrieved by name. it's `static`, after all.